app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,41 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def create
+    Merchant.create(merchant_params)
+    redirect_to '/merchants'
+  end
+
+  def new
+    @merchant = Merchant.new
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    @merchant.update(merchant_params)
+    @merchant.save
```

Comment:
I don't think you need `.save` here since update tries to save to the DB for you